subprojects/testing-base/src/main/java/org/gradle/api/tasks/testing/AbstractTestTask.java

```diff
@@ -102,11 +105,53 @@
  */
 @DisableCachingByDefault(because = "Abstract super-class, not to be instantiated directly")
 public abstract class AbstractTestTask extends ConventionTask implements VerificationTask, Reporting<TestTaskReports> {
+
+    /**
+     * Wraps a list of listeners to subscribe, and lazily configures an anonymous broadcaster with those listeners when requested.
+     * Instances of this class are suitable for serialization (as long as listeners are serializable as well).
```

Comment:
💭 Why don't we have a marker interface `CCSerializable` or something similar to indicate what types can and can't be serialized with the configuration cache?  

Then we could use indicate whether a type (including `BroadcastSubscriptions` here) can be serialized via the class hierarchy and be able to check all the listeners added to it at runtime.

(I imagine there must be a reason, I just don't know what it is.)

Reply:
I guess that would work in many cases. One downside would be to make the CC concern more in-your-face, while as it stands now, CC-compatible code does not have to have any CC-types in their dependencies.